### PR TITLE
Implement session cleanup

### LINKED ISSUE #52: Implement session cleanup
`xmantissa.websession` claims that expired sessions will be cleaned every `SESSION_CLEAN_FREQUENCY` seconds. In fact, this isn't true, nothing cares about that value, and nothing ever deletes a persistent session except for explicitly logging out.

### PR DESCRIPTION
**Pull request by mithrandi**
Fixes #52.

I did a lot of cleanup in `xmantissa.websession`; to make this easier to review, I tried to separate my commits into changesets that can be reviewed individually.

### CODE REVIEW

**Comment by mithrandi** [comment]
PR diff coverage is now at 100%.

File: xmantissa/websession.py [hunk 1]
```diff
@@ -241,20 +301,39 @@ def savorSessionCookie(self, request):
         """
         Make the session cookie last as long as the persistant session.
```
Comment (jonathanj) [inline xmantissa/websession.py]: Another spelling error of "persistent".

File: xmantissa/websession.py [hunk 2]
```diff
@@ -277,16 +356,23 @@ def cbLoginSuccess(input):
...
 
-        return guard.SessionWrapper.login(self, request, session, creds, segments
-                                          ).addCallback(cbLoginSuccess)
+        return (
+            guard.SessionWrapper.login(
+                self, request, session, creds, segments)
+            .addCallback(cbLoginSuccess))
 
 
     def explicitLogout(self, session):
```
Comment (jonathanj) [inline xmantissa/websession.py:365]: There don't seem to be any tests (not in `test_websession` anyway) for the explicit log out case. Did I miss these somewhere?

File: xmantissa/websession.py [hunk 3]
```diff
@@ -241,20 +301,39 @@ def savorSessionCookie(self, request):
...
+
+        @type creds: L{twisted.cred.credentials.ICredentials}
+        @param creds: The credentials the user presented.
+
+        @type segments: L{tuple}
+        @param segments: The remaining segments of the URL.
+
+        @return: A deferred firing with the user's avatar.
         """
+        self._maybeCleanSessions()
```
Comment (jonathanj) [inline xmantissa/websession.py:335]: I don't think this behaviour is covered by a test case, namely that logging in causes old enough sessions to be cleaned.

Comment (mithrandi) [inline xmantissa/websession.py:335]: The case you mentioned is covered by `test_sessionCleanup`, did you mean something else?

File: xmantissa/websession.py (hunk 2 above)
Comment (mithrandi) [inline xmantissa/websession.py:365]: [Codecov confirms that it's not covered](https://codecov.io/github/twisted/mantissa/xmantissa/websession.py?ref=b103aa11b20ffe6667e6a0eb0e1df5fa04168708#l-377). I didn't change the code in this branch, but I guess I may as well add a test while I'm here.

File: xmantissa/websession.py (hunk 1 above)
Comment (mithrandi) [inline xmantissa/websession.py]: Gah! Thanks, fixed (as well as another one that searching found for me).

File: xmantissa/websession.py (hunk 2 above)
Comment (mithrandi) [inline xmantissa/websession.py:365]: Added.

File: xmantissa/websession.py (hunk 3 above)
Comment (mithrandi) [inline xmantissa/websession.py:335]: I expanded the test coverage anyway, so maybe I caught whatever it was you meant.

**Comment by jonathanj** [comment]
Looks good to me, please merge.

**Pull request merged**